packages/react-ui/src/hooks/useImperativeIntl.js

```diff
@@ -0,0 +1,38 @@
+import { IntlContext, createIntl, createIntlCache, useIntl } from 'react-intl';
+import { messages } from '../localization';
+import { useContext, useMemo } from 'react';
+import { match } from '@formatjs/intl-localematcher';
+
+export const DEFAULT_LOCALE = 'en-US';
+const cache = createIntlCache();
+
+// TODO: Pending to get external intl provider
+
+export default function useImperativeIntl() {
+  const extenal = useIntl();
```

Comment:
typo 'external'?

Reply:
it shouldn't be here actually:

```
const extenal = useIntl();
const externalIntl = useContext(IntlContext);
```

take same object, but first asserts (and throws!) when context is null, so this is undesirable, we would like to leave as much backwards compatibility as needed and make ReactIntlProvider optional ...

second, just gets value - if absent, it falls back to null and we can use our defaults